### DISCUSSION TITLE
Using createLogger with Nebula Managed Package?

### DISCUSSION THREAD

**Discussion by initdotd** [jongpie/NebulaLogger#528] [Q&A] [answered]
Hi All,

I'm trying to use the LWC logger from the managed package installation in a custom LWC.  I feel like I'm missing something simple on how to reference the component via the import.  I've tried a variety of different approaches and all result in the "No MODULE named..." error at compile/push to org time.

Ultimately, I'm unable to leverage Nebula via LWC when using the Nebula managed package.  

```
import {LightningElement} from 'lwc';
import { createLogger } from 'nebula_logger'; // This doesn't work with managed package installation of Nebula
/*
I've tried referencing the component several ways with no luck.  I was able to use the legacy approach <nebula_logger></nebula_logger>, but that then fails to work with the querySelector in the javascript controller.
*/

export default class LoggerDemo extends LightningElement {

    logger = createLogger();

    constructor() {
        super();
        this.logger.info('Hello, world!');
        this.logger.saveLog();
    }
}
```

Any insight or advice is greatly appreciated and thanks in advance!

**Comment by jefersonchaves** [comment]
Hi Jeff,

Have you tried `namespace/module` like `import { createLogger } from 'nebula/logger';`?

That is the preferred way based on [LWC docs](https://developer.salesforce.com/docs/platform/lwc/guide/create-components-namespace.html).

**Comment by jongpie** [comment]
@initdotd this is now officially a bug with the managed package - see [this comment/issue](https://github.com/jongpie/NebulaLogger/issues/533#issuecomment-1730007405) for more details. I hope to have the issue resolved in the next release of the managed package, `v4.12.0`.

Once the fix is released, you'll be able to import the component using `import { createLogger } from 'Nebula/logger';`.

**Reply by initdotd** [reply]
Thanks and that is great news!

**Reply by jongpie** [reply, marked as answer]
@initdotd the `logger` LWC is now available in the managed package in [release `v4.13.0`](https://github.com/jongpie/NebulaLogger/releases/tag/v4.13.0)!

**Reply by initdotd** [reply]
@jongpie Excellent and thank you!
